### DISCUSSION TITLE
Conditionally revealing inputs based on another input in a Framework Doc

### DISCUSSION THREAD

**Discussion by jack-davison** [observablehq/framework#1673] [Q&A] [answered]
Good morning,

Apologies if this is a very simple question, but I'm curious if it's possible to conditionally show inputs to the user based on the value of another input? I've had a look through the documentation but can't quite find anything that does the job.

For example, let's say I have `category` and `value`, as outlined below.

```js
const category = view(Inputs.select(["A", "B", "C"]))

const value = view(Inputs.range([1,100]))
```

If I only want the range Input `value` to be revealed to the user when `category` is `"A"` (perhaps it defaults to some value otherwise, like `0` or `null`), how would I do that? When it is `"B"` or `"C"` I don't want it visible at all.

I know I could disable it, or set it to only take one value, but I worry that will end up looking messy if there are multiple "branches" of options.

What I'm after is effectively an analogue to `conditionalPanel()` in `{shiny}`: https://shiny.posit.co/r/reference/shiny/1.4.0/conditionalpanel

Thank you!

**Comment by Fil** [comment, marked as answer]
Sure. You could name the input and display it conditionally:

~~~md

```js
const category = view(Inputs.select(["A", "B", "C"]))
const valueInput = Inputs.range([1,100])
const value = Generators.input(valueInput);
```

${category === "A" ? valueInput : ""}

~~~

**Reply by jack-davison** [reply]
Easy as that! Thanks very much; I'll read up on `Generators.input()`

**Reply by Fil** [reply]
You probably already have found the [documentation](https://observablehq.com/framework/lib/generators#input-element).
